### PR TITLE
Add detection of length input parameters

### PR DESCRIPTION
**Pull request by EPashkin**
Part of https://github.com/gtk-rs/gtk/issues/477 and https://github.com/gtk-rs/gir/issues/264.

### CODE REVIEW

File: src/analysis/function_parameters.rs [hunk 1]
```diff
@@ -311,3 +312,62 @@ fn get_length_type(
...
+    );
+    array.map(|p| &p.name)
+}
+
+fn is_length(par: &library::Parameter) -> bool {
+    if par.direction != library::ParameterDirection::In {
+        return false;
+    }
+
+    if par.name.len() >= 3 && &par.name[0..3] == "len" {
```
Comment (sdroege) [inline src/analysis/function_parameters.rs]: Usually it's ending with 'len', not beginning. 'objects_len', 'string_len', etc. Or with the full word 'length'.

**Comment by sdroege** [comment]
This needs a new configuration so it can be disabled. There will be false positives otherwise, where an argument is detected as a length parameter but actually is not.

File: src/analysis/function_parameters.rs (hunk 1 above)
Comment (EPashkin) [inline src/analysis/function_parameters.rs]: Changing this to `if len >= 3 && &par.name[len-3..len] == "len" {`
don't change any functions.
As it don't remove any too and sound reasonable I use new variant.

**Comment by EPashkin** [comment]
Something like `dont_detect_lengths=true` on function level?

**Comment by EPashkin** [comment]
or "detect_lengths=false"?

**Comment by sdroege** [comment]
Yes, either name seems ok

**Comment by EPashkin** [comment]
Added in as "disable_length_detect=true"

**Comment by EPashkin** [comment]
Checked all other crates: PRs needed for glib, gio, pango but as it contains only generated code I postpone it until this merged.

**Pull request merged**